src/acl.c

```diff
@@ -447,19 +486,45 @@ int ACLCountCategoryBitsForUser(user *u, unsigned long *on, unsigned long *off,
     if (!cflag) return C_ERR;
 
     *on = *off = 0;
-    dictIterator *di = dictGetIterator(server.orig_commands);
+    ACLCountCategoryBitsForUserLogic(server.orig_commands, u, on, off, cflag);
+    return C_OK;
+}
+
+sds ACLDescribeUserCommandRulesSingleCommands(user *u, user *fakeuser, sds rules, dict *commands) {
+    /* Fix the final ACLs with single commands differences. */
+    dictIterator *di = dictGetIterator(commands);
     dictEntry *de;
     while ((de = dictNext(di)) != NULL) {
         struct redisCommand *cmd = dictGetVal(de);
-        if (cmd->flags & cflag) {
-            if (ACLGetUserCommandBit(u,cmd->id))
-                (*on)++;
-            else
-                (*off)++;
+        int userbit = ACLGetUserCommandBit(u,cmd->id);
+        int fakebit = ACLGetUserCommandBit(fakeuser,cmd->id);
+        if (userbit != fakebit) {
+            rules = sdscatlen(rules, userbit ? "+" : "-", 1);
+            sds fullname = getFullCommandName(cmd);
+            rules = sdscat(rules,fullname);
+            sdsfree(fullname);
+            rules = sdscatlen(rules," ",1);
+            ACLChangeCommandPerm(fakeuser,cmd,userbit);
+        }
+
+        if (cmd->subcommands_dict)
+            rules = ACLDescribeUserCommandRulesSingleCommands(u,fakeuser,rules,cmd->subcommands_dict);
+
+        /* Emit the subcommands if there are any. */
```

Comment:
@madolson can you please take a closer look at this one? 
Maybe requires more testing or test code

Reply:
More testing yeah, I think it looks correct.

Reply:
@guybe7 please add some tests for ACL GETUSER with sub-commands, and commands / sub-commands with firstarg